Doc/library/stdtypes.rst

```diff
@@ -2392,6 +2392,196 @@ that ``'\0'`` is the end of the string.
 .. index::
    single: buffer protocol; binary sequence types
 
+
+.. index::
+   single: formatted string literal
+   single: interpolated string literal
+   single: string; formatted literal
+   single: string; interpolated literal
+   single: f-string
+   single: {} (curly brackets); in formatted string literal
+   single: ! (exclamation); in formatted string literal
+   single: : (colon); in formatted string literal
+   single: = (equal); in debug string literal
+.. _f-string-formated-string-literal:
+
+``f-string``-Formatted String Literal
+-------------------------------------
+.. versionadded:: 3.6
+
+A :dfn:`formatted string literal` or :dfn:`f-string` is a string literal
+that is prefixed with ``'f'`` or ``'F'``.  These strings may contain
+replacement fields, which are expressions delimited by curly braces ``{}``.
+While other string literals always have a constant value, formatted strings
+are really expressions evaluated at run time.
+
+This adds flexibility to how strings can be formatted::
+
+   >>> name = "Fred"
+   >>> f"He said his name is {name!r}."
+   "He said his name is 'Fred'."
+
+The above is an example and will become clear as you read below.
+
+Everything outside the curly braces is treated as literally with the exception
+of double braces which are treated as one brace::
+
+   >>> f"This is a string"
+   'This is a string.'
+   >>> f'{{'
+   '{'
+   >>> F"{{"
+   '}'
+   >>> F'{{ }}'
+   '{ }'
+
+A single opening curly bracket, ``'{'``, marks a replacement field that starts with
+a python expression.::
+
+   >>> f"4 * 3 is {4 * 3}"
+   '4 * 3 is 12'
+
+The above is quite contrived and a more useful use in code could be::
+
+   >>> a = 4
+   >>> b = 3
+   >>> f"The product of {a} and {b} is {a * b}"
+   'The product of 4 and 3 is 12'
+
+Or more simply::
+
+   >>> a = 4
+   >>> b = 3
+   >>> f"The product is {a * b}"
+   'The product is 12'
```

Comment:
I don't think this adds anything to the previous example.

Reply:
Removed.